wnfs/src/private/directory.rs

```diff
@@ -616,6 +616,94 @@ impl PrivateDirectory {
         }
     }
 
+    /// Opens a mutable reference to the specified file.
+    /// If the file is missing, it initializes an empty file and give a mut reference to that.
+    /// If the file already exists, it will copy it to the next revision, update the edit time, and give a mut reference to that.
+    /// # Examples
+    /// ```
+    /// use std::rc::Rc;
+    /// use chrono::Utc;
+    /// use rand::thread_rng;
+    /// use wnfs::{
+    ///    private::{PrivateForest, PrivateRef, PrivateDirectory},
+    ///    common::{BlockStore, MemoryBlockStore},
+    ///    namefilter::Namefilter,
+    /// };
+    /// #[async_std::main]
+    /// async fn main() {
+    ///    let store = &mut MemoryBlockStore::default();
+    ///    let rng = &mut thread_rng();
+    ///    let mut forest = &mut Rc::new(PrivateForest::new());
+    ///    let root_dir = &mut Rc::new(PrivateDirectory::new(
+    ///         Namefilter::default(),
+    ///         Utc::now(),
+    ///         rng,
+    ///    ));
+    ///    let content = b"print('hello world')";
+    ///    root_dir
+    ///        .write(
+    ///            &["code".into(), "hello.py".into()],
+    ///            true,
+    ///            Utc::now(),
+    ///            content.to_vec(),
+    ///            &mut forest,
+    ///            store,
+    ///            rng
+    ///        )
+    ///        .await
+    ///        .unwrap();
+    ///    let mut file = {
+    ///        root_dir
+    ///            .open_file_mut(&["code".into(), "hello.py".into()], true, Utc::now(), forest, store, rng)
+    ///            .await
+    ///            .unwrap()
+    ///    };
```

Comment:
I think you can remove the extra scope here.
```suggestion
    ///    let mut file = root_dir
    ///        .open_file_mut(&["code".into(), "hello.py".into()], true, Utc::now(), forest, store, rng)
    ///        .await
    ///        .unwrap();
```